File: Data Structures & Algorithms/daily-temperatures/submission-1.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        std::vector<int> result (temperatures.size());
        std::stack<std::pair<int, int>> st{};

        for (int i = 0; i < temperatures.size(); i++) {
            if (st.empty() || st.top().first >= temperatures[i]) {
                st.push(std::make_pair(temperatures[i], i));
                continue;
            }

            while (!st.empty() && st.top().first < temperatures[i]) {
                int days = i - st.top().second;
                result.at(st.top().second) = days;
                st.pop();
            }

            st.push(std::make_pair(temperatures[i], i));
        }

        return result;
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-1.cs
public class Solution {
    public bool IsValid(string s) {
        var stack = new Stack<char>();

        foreach (char c in s) {
            if (c == '(' || c == '[' || c == '{') {
                stack.Push(c);
                continue;
            }

            if (stack.Count == 0) return false;

            char top = stack.Pop();
            if ((c == ')' && top != '(') 
                || (c == ']' && top != '[') 
                || (c == '}' && top != '{')) 
                return false;
        }

        return stack.Count == 0;
    }
}
